DSharpPlus/Clients/DiscordClient.WebSocket.cs

```diff
@@ -404,14 +404,32 @@ internal async Task SendHeartbeatAsync(long seq)
         {
             var more_than_5 = Volatile.Read(ref this._skippedHeartbeats) > 5;
             var guilds_comp = Volatile.Read(ref this._guildDownloadCompleted);
+
             if (guilds_comp && more_than_5)
             {
                 this.Logger.LogCritical(LoggerEvents.HeartbeatFailure, "Server failed to acknowledge more than 5 heartbeats - connection is zombie");
+
+                var args = new ZombiedEventArgs
+                {
+                    Failures = Volatile.Read(ref this._skippedHeartbeats),
```

Comment:
Does this need to be `Volatile.Read`?

Reply:
> 
> 
> Does this need to be `Volatile.Read`?

I only made it so as the other reads are also volatile. If you believe it should not be I can change that

Reply:
If for the sake of consistency, sure